include/sound/sof/topology.h

```diff
@@ -218,21 +218,27 @@ struct sof_ipc_comp_reply {
  * Pipeline
  */
 
+/** \brief Types of pipeline scheduling time domains */
+enum sof_ipc_pipe_sched_time_domain {
+	SOF_TIME_DOMAIN_DMA = 0,	/**< DMA interrupt */
+	SOF_TIME_DOMAIN_TIMER,		/**< Timer interrupt */
+};
+
 /* new pipeline - SOF_IPC_TPLG_PIPE_NEW */
 struct sof_ipc_pipe_new {
 	struct sof_ipc_cmd_hdr hdr;
 	uint32_t comp_id;	/**< component id for pipeline */
 	uint32_t pipeline_id;	/**< pipeline id */
 	uint32_t sched_id;	/**< Scheduling component id */
 	uint32_t core;		/**< core we run on */
-	uint32_t deadline;	/**< execution completion deadline in us*/
+	uint32_t period;	/**< execution period in us*/
 	uint32_t priority;	/**< priority level 0 (low) to 10 (max) */
 	uint32_t period_mips;	/**< worst case instruction count per period */
 	uint32_t frames_per_sched;/**< output frames of pipeline, 0 is variable */
 	uint32_t xrun_limit_usecs; /**< report xruns greater than limit */
 
-	/* non zero if timer scheduled, otherwise DAI DMA irq scheduled */
-	uint32_t timer_delay;
+	/**< scheduling time domain */
+	enum sof_ipc_pipe_sched_time_domain time_domain;
```

Comment:
I thought we agreed enums are not portable and need to be avoided in IPC structures?

Reply:
@plbossart Fixed.